### ISSUE TITLE
styleActiveLine option breaks arrow key movement on iPad Pros

### ISSUE THREAD

**Issue by vincentwoo** [codemirror/CodeMirror#4194]
*I know that full mobile support is not an explicit goal of this project, but I think iPad pros are pretty close to desktops in some ways and that this particular bug is surprising enough to warrant a look.*

This bug is pretty easy to duplicate and kind of surprising.

1. Open an IOS simulator or a physical iPad if you have one.
2. Navigate in mobile safari to https://codemirror.net/demo/activeline.html
3. Attempt to use the up arrow. It won't work (the cursor kind of vibrates and stays in place). The left arrow won't be able to go through the line boundary, either.

**Comment by vincentwoo** [comment]
[This](https://github.com/codemirror/CodeMirror/blob/master/addon/selection/active-line.js#L60) `cm.operation` call in active-line.js appears to be the cause of the problem - for some reason, trying to change the css classes on the active line in the middle of the selection change messes up the arrow key movement.

Interestingly, I found that if you put a `setTimeout` on the operation, you can move the cursor normally as long as the highlighted active line has not yet caught up with the actual active line.

**Comment by kofifus** [comment]
hey vincent .. what does up arrow means on ipad ? touch above ?

**Comment by vincentwoo** [comment]
If you use an ipad pro with hardware keyboard on you can literally use a keyboard. In this case, I am talking about a normal up/down arrow that appears on a keyboard.

**Comment by kofifus** [comment]
ah ok didn't know you can do that :)

**Comment by marijnh** [comment]
That is very strange. Unfortunately, I can't directly test it, not having an Apple machine to run an emulator on, and not having a keyboard for my iPad.

One thing that might be happening is that the display update somehow ends up messing with Safari's selection in a problematic way. But I don't really see why that would happen.

**Comment by vincentwoo** [comment]
Yeah, I ended up just disabling that addon for mobile users. I'm not sure how to proceed so I'm happy to just leave this issue here as a warning to future users.

**Comment by jaypeedevlin** [comment]
@marijnh I have used browserstack's ipad emulator as a stand-in for these sort of things.

Interestingly when I load https://codemirror.net/demo/activeline.html I can get all arrow keys to work *except* up, except where the cursor is on the left hand end of the line, where the left arrow key would move the up to the line above anyway.

**Comment by marijnh** [comment]
Could you see if attached patch helps at all here?

**Comment by vincentwoo** [comment]
Will try hopefully in the coming days, need to get on an OSX machine w/ simulator! Anyone else is welcome to try too (using browserstack et al).

**Comment by idevMike** [comment]
The up arrow seems to work for me, however the back arrow doesn't go to the previous line when you are at the beginning of the next line.
